Lib/logging/handlers.py

```diff
@@ -827,11 +828,26 @@ def __init__(self, address=('localhost', SYSLOG_UDP_PORT),
             self.unixsocket = False
             if socktype is None:
                 socktype = socket.SOCK_DGRAM
-            self.socket = socket.socket(socket.AF_INET, socktype)
-            if socktype == socket.SOCK_STREAM:
-                self.socket.connect(address)
-            self.socktype = socktype
-        self.formatter = None
+            host, port = address
+            err = None
+            for res in socket.getaddrinfo(host, port, 0, socktype):
+                af, socktype, proto, cannonname, sa = res
+                sock = None
+                try:
+                    sock = socket.socket(af, socktype, proto)
+                    if socktype == socket.SOCK_STREAM:
+                        sock.connect(sa)
+                    self.socket = sock
+                    self.socktype = socktype
+                    return
+                except OSError as _:
```

Comment:
Honestly speaking, I copied most logic from `socket.create_connection()`. But I don't understand how to use `finally` here. We only need to close the socket when `socket.connect()` fails.